corva/models/stream.py

```diff
@@ -1,27 +1,55 @@
-from typing import Any, Dict, List, Optional
+from typing import Dict, List, Optional
 
-from corva.models.base import BaseContext, BaseEventData
+from corva.models.base import BaseContext, BaseEventData, ListEvent
 from corva.state.redis_state import RedisState
 
 
 class StreamContext(BaseContext):
     state: RedisState
 
 
-class Record(BaseEventData):
-    timestamp: Optional[int] = None
-    asset_id: int
-    company_id: int
-    version: int
-    data: Dict[str, Any]
-    measured_depth: Optional[float] = None
-    collection: str
+class StreamEventData(BaseEventData):
+    class Record(BaseEventData):
+        class Data(BaseEventData):
+            hole_depth: Optional[float] = None
```

Comment:
wow, let's not do nested classes. Especially triple folding of these

Reply:
I think these can be unpacked as their own independent classes and then set as attributes on the class they belong to.

Reply:
unnested the classes